packages/core/lib/package.js

```diff
@@ -210,169 +209,136 @@ const Package = {
-      if (err) {
-        if (callbackPassed) {
-          return callback(err);
-        }
-        throw err;
+    let result;
+    try {
+      await Networks.asURIs(options, deployed_networks);
+    } catch (error) {
+      if (callbackPassed) {
+        return callback(err);
       }
+      throw err;
+    }
 
-      var uris = result.uris;
-
-      if (result.failed.length > 0) {
-        const message =
-          "Could not connect to the following networks: " +
-          result.failed.join(", ") +
-          ". These networks have deployed " +
-          "artifacts that can't be published as a package without an active " +
-          "and accessible connection. Please ensure clients for each " +
-          "network are up and running prior to publishing, or use the -n " +
-          "option to specify specific networks you'd like published.";
-        if (callbackPassed) {
-          return callback(new Error(message));
-        }
-        throw new Error(message);
+    var uris = result.uris;
+
+    if (result.failed.length > 0) {
+      const message =
+        "Could not connect to the following networks: " +
+        result.failed.join(", ") +
+        ". These networks have deployed " +
+        "artifacts that can't be published as a package without an active " +
+        "and accessible connection. Please ensure clients for each " +
+        "network are up and running prior to publishing, or use the -n " +
+        "option to specify specific networks you'd like published.";
+      if (callbackPassed) {
+        return callback(new Error(message));
       }
+      throw new Error(message);
+    }
 
-      var files = fs.readdirSync(options.contracts_build_directory);
-      files = files.filter(file => file.endsWith(".json"));
+    var files = fs.readdirSync(options.contracts_build_directory);
+    files = files.filter(file => file.endsWith(".json"));
 
-      if (!files.length) {
-        const message =
-          "Could not locate any publishable artifacts in " +
-          options.contracts_build_directory +
-          ". " +
-          "Run `truffle compile` before publishing.";
-        if (callbackPassed) {
-          return callback(new Error(message));
-        }
-        throw new Error(message);
+    if (!files.length) {
+      const message =
+        "Could not locate any publishable artifacts in " +
+        options.contracts_build_directory +
+        ". " +
+        "Run `truffle compile` before publishing.";
+      if (callbackPassed) {
+        return callback(new Error(message));
       }
+      throw new Error(message);
+    }
 
-      var promises = files.map(file => new Promise((resolve, reject) => {
-        fs.readFile(
-          path.join(options.contracts_build_directory, file),
-          "utf8",
-          function(err, body) {
-            if (err) return reject(err);
+    const contracts = files.map(file => {
+      const body = fs.readFileSync(
```

Comment:
Do we want to do this synchronously?

Reply:
This is another case where I don't think we gain much by having this be asynchronous. I don't really mind it being async but I like this for it's clarity rather than adding a bunch of junk to deal with asynchronous methods. There will be cases where the time saved will be meaningful but it seemed to me that it wouldn't matter much in this scenario. Feel free to disagree with me.

Reply:
In this case we are looping through a list of contracts and doing file IO where subsequent IO doesn't depend on previous IO.  This just beckons for async - I would delegate this off to libuv and call it a day.

Reply:
Ok, changed back to an ugly asynchronous operation :)